src/core/HookManager.sol

```diff
@@ -22,14 +22,24 @@ abstract contract HookManager {
     error HookPostCheckFailed();
     error HookAlreadyInstalled(address currentHook);
 
-    modifier withHook() {
+    function _preCheck() internal returns (address hook, bytes memory hookData) {
         address hook = _getHook();
```

Comment:
since type is already declared in return 
change this 
address hook -> hook
same with hookData

Reply:
good shout, done